lib/util/flattenOperands.js

```diff
@@ -34,293 +35,80 @@ interested in how that works
 // 2+2+2, ie one + node that has three children.
 // Input: an expression tree
 // Output: the expression tree updated with flattened operations
-function flattenOperands(node) {
-  if (Node.Type.isConstant(node, true)) {
-    // the evaluate() changes unary minuses around constant nodes to constant nodes
-    // with negative values.
-    const constNode = Node.Creator.constant(evaluate(node));
-    if (node.changeGroup) {
-      constNode.changeGroup = node.changeGroup;
-    }
-    return constNode;
-  }
-  else if (Node.Type.isOperator(node)) {
-    if ('+-/*'.includes(node.op)) {
-      let parentOp;
-      if (node.op === '/') {
-        // Division is flattened in partner with multiplication. This means
-        // that after collecting the operands, they'll be children args of *
-        parentOp = '*';
-      }
-      else if (node.op === '-') {
-        // Subtraction is flattened in partner with addition, This means that
-        // after collecting the operands, they'll be children args of +
-        parentOp = '+';
+function flattenOperands(root) {
+  return replace(root, {
+    leave(node) {
+      if (query.isAdd(node)) {
+        return build.add(
+          ...node.args.reduce((accum, arg) => {
+            if (query.isAdd(arg)) {
+              return accum.concat(arg.args);
+            }
+            else {
+              return accum.concat(arg);
+            }
+          }, [])
+        );
       }
-      else {
-        parentOp = node.op;
+      else if (query.isMul(node)) {
+        // Rewrite explicit multiplication to be implicit if the factors look
+        // like those of a polynomial.  The first factor must be a number and
+        // the following factors must not be numbers.
+        // e.g. 2 * x^2 * y -> 2 x^2 y
+        if (isPolynomialTerm(node) && query.isNumber(node.args[0])
```

Comment:
I left the old one b/c I wanted to keep the scope of this diff limited.  As we convert more of the code to use `math-rules` we'll either rewrite the original one or use this one.  It's possible that we may want two different behaviors.  If that is that case we'll have to come up with new name(s).  I'll add a comment by the definition of this new one.